library/src/main/java/io/voucherify/android/client/module/VoucherModule.java

```diff
@@ -17,25 +18,16 @@
      * @param trackingId custom tracking id to track voucher consumers
      */
     public VoucherModule(VoucherifyApi api, Scheduler scheduler, String trackingId) {
-        super(api, scheduler, trackingId);
+        this.validation = new Validation(api, scheduler, trackingId);
```

Comment:
Maybe it would be better to pass Validation and Redemption as params and don't use 'new' inside the constructor as we will have problems with mocking during testing.